### DISCUSSION TITLE
Removing restriction in `create_model` of not being able to specify both the `__config__` and the `__base__`

### DISCUSSION THREAD

**Discussion by robcxyz** [pydantic/pydantic#5782] [Question]
When using `create_model`, it throws an error when you specify both the `__base__` and `__config__` in this code: 

https://github.com/pydantic/pydantic/blob/334e47b51eb0b0d75d5aef884aaa6d92292a2370/pydantic/main.py#L1275-L1280

Which I don't quite understand as when I remove the error in the `create_model` function I can specify both the `__base__` and `__config__` fine like so: 

```python
from pydantic import BaseModel, create_model


class TheBaseModel(BaseModel):
    foo: str = "bar"


NewModel = create_model(
    "Foo",
    __base__=TheBaseModel,
    __config__={
        'extra': 'allow'
    },
    **{'stuff': (str, 'things')},
)

new_model = NewModel(baz=1)

assert new_model.baz
```

Which works fine. 

Also only one test fails, the one checking for the error, with the error removed: 

https://github.com/pydantic/pydantic/blob/334e47b51eb0b0d75d5aef884aaa6d92292a2370/tests/test_create_model.py#L80-L82

So my question is why exactly is this error here as it doesn't seem to be needed. For my particular use case, I need to be able to specify both the config and base when building the model. Another issue (#2137) describes a way to compose a model from two other models but I simply want to inject the config as a dict, not from another model. 

Happy to push PR updating this with tests.

**Comment by zmievsa** [comment]
@samuelcolvin Hi! Sorry for pinging you directly but this discussion has been unanswered for more than a year now so I was wondering whether we could raise it again and understand the rationale for why this behavior is prohibited.
